zerver/actions/message_send.py

```diff
@@ -1602,7 +1603,10 @@ def internal_prep_private_message(
     See _internal_prep_message for details of how this works.
     """
     addressee = Addressee.for_user_profile(recipient_user)
-    realm = recipient_user.realm
+    if not is_cross_realm_bot_email(recipient_user.delivery_email):
+        realm = recipient_user.realm
+    else:
+        realm = sender.realm
```

Comment:
I don't think bug should have affected anything in production, because we don't use `internal_prep_private_message` to send messages from a normal user to a cross-realm bot. But probably worth checking anyway (as well as to find any other potential bugs) by looking up `Message`s with `realm="zulipinternal"` where the `sender` is in another realm (that would be a clearly bugged message)

Reply:
Yeah, that seems like a thing we can check after merging this.